webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/ParametersTable.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { TableComposable, Thead, Tbody, Tr, Th } from '@patternfly/react-table';
+import {
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+  Button,
+  Spinner,
+  SearchInput,
+} from '@patternfly/react-core';
+import { translate as __ } from '../../../../common/I18n';
+import { ParametersTableRow } from './TableRow';
+import { columnNames, HOST_PARAM } from './ParametersConstants';
+import Pagination from '../../../Pagination';
+import { useForemanSettings } from '../../../../Root/Context/ForemanContext';
+import './Parameters.scss';
+import { STATUS } from '../../../../constants';
+
+export const ParametersTable = ({
+  status,
+  hostId,
+  allParameters,
+  editHostsPermission,
+}) => {
+  const [editingRow, setEditingRow] = useState(-1);
+  const [activeSortDirection, setActiveSortDirection] = useState('asc');
+  const { perPage: settingsPerPage = 20 } = useForemanSettings() || {};
+  const [page, setPage] = useState(1);
+  const [perPage, setPerPage] = useState(settingsPerPage);
+  const [showNewRow, setShowNewRow] = useState(false);
+  const [search, setSearch] = useState('');
+
+  const sortedParams = (
+    direction = activeSortDirection,
+    params = allParameters
+  ) =>
+    [...params].sort((a, b) => {
+      if (direction === 'asc') {
+        return a.name.localeCompare(b.name);
+      }
+      return b.name.localeCompare(a.name);
+    });
+  const [parameters, setParameters] = useState(
+    sortedParams().slice(0, settingsPerPage)
+  );
+  const sortParams = {
+    sortBy: {
+      index: 0,
+      direction: activeSortDirection,
+      defaultDirection: activeSortDirection,
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortDirection(direction);
+      setParameters(sortedParams(direction).slice(0, perPage));
+    },
+    columnIndex: 0,
+  };
```

Comment:
In Katello we have a custom hook [`useTableSort`](https://github.com/Katello/katello/blob/723f67e87aa8d482be64a8e7dda054f93f98ae96/webpack/components/Table/TableHooks.js#L270) which we can reuse anywhere. Duplicating that in Foreman isn't ideal, but writing it inside a single specific component is worse. What if you extract this to a hook, so at least other Foreman components can use table sorting easily?  And maybe even Katello could use it if it's stable and adaptable enough.

Reply:
Took Katelloes `useTableSort` and added an arg for `onSort`